apps/src/blockly/addons/cdoFieldImageDropdown.js

```diff
@@ -1,9 +1,36 @@
-import CdoFieldDropdown from './cdoFieldDropdown';
+import {FieldGridDropdown} from '@blockly/field-grid-dropdown';
+import color from '@cdo/apps/util/color';
 
-export class CdoFieldImageDropdown extends CdoFieldDropdown {
+// Note that this class *does not* inherit from CdoFieldDropdown
+export class CdoFieldImageDropdown extends FieldGridDropdown {
   constructor(menuGenerator, width, height, buttons) {
-    const fixedMenuGenerator = fixMenuGenerator(menuGenerator, width, height);
-    super(fixedMenuGenerator);
+    // We have to decide how many columns to have when we create the block. The
+    // number of options in the block can change over time, but we can just use
+    // the initial number of options to determine how many columns to show- we're
+    // assuming that the number of options usually won't change that drastically,
+    // so the number of columns can probably stay the same.
+    const initialOptions = fixMenuGenerator(menuGenerator, width, height);
+    const numColumns =
+      initialOptions.length <= 7
+        ? 1
+        : Math.floor(Math.sqrt(initialOptions.length));
+
+    super(
+      () => fixMenuGenerator(menuGenerator, width, height),
+      undefined /* validator */,
+      {columns: numColumns}
+    );
+  }
+
+  showEditor_(e = undefined) {
```

Comment:
if blockly core plans to support the color configuration we want, should we add a comment here explaining that we can remove this override when that config option is added?